### PR TITLE
Decouple pbs_benchpress from PBSTestSuite to CommonTestSuite

### PR DESCRIPTION
**Pull request by varunsonkar**
#### Bug/feature Description
Decouple pbs_benchpress from PBSTestSuite to CommonTestSuite

#### Affected Platform(s)
None

#### Cause / Analysis / Design
As of now pbs_benchpress is tightly coupled with PBSTestsuite, and thus require PBS to be pre-installed for running any test case. The issue is that we have PBS Installation test cases that require overwriting of PBSTestSuite when automated with current design. So we are proposing to decouple pbs_bencpress from PBSTestSuite to CommonTestSuite, so that such kind of test cases that do not require PBS to be pre-installed can be easily automated.

#### Solution Description
* *The solution is to introduce new class "CommonTestSuite" that inherits from "unittest.TestCase" and make PBSTestSuite as subclass of CommonTestSuite.
With this changes pbs_benchpress, ptl_test_info.py, ptl_test_loader.py, ptl_test_runner.py has to be modified accordingly as these scripts look for "PBSTestSuite" for loading and running PTL test cases.*

#### Testing logs/output
Smoke Test logs after making the changes:
[Smoke_Test_CommonTestSuite.log](https://github.com/PBSPro/pbspro/files/2169123/Smoke_Test_CommonTestSuite.log)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by agrawalravi90** [comment]
"The solution is to introduce new class "CommonTestSuite" that inherits from "unittest.TestCase" and make PBSTestSuite as subclass of CommonTestSuite."

Why not just use unittest as the common test framework ?

**Comment by varunsonkar** [comment]
@agrawalravi90 ,
This was discussed with @crjayadev , @hirenvadalia , @kjakkali  and @vishwaks . The idea is that something which is related to the generic PTL test framework, for e.g.:
PR: #728 : "PP-1278: Need a skip decorator in PTL which will skip the test withou…"
is desired to go in "CommonTestSuite". 
Moreover the above mentioned "skip decorator" is not a utility. It is the extended functionality of PTL test Framework. So we believe the right place for it to go is the "CommonTestSuite".

**Comment by agrawalravi90** [comment]
Thanks for more details Varun.

I think it'll help if you could explain your use case a bit more. Please change the title of the PR (and the commit message) to the actual issue that this change is going to fix (as opposed to the solution) and clarify how CommonTestSuite is going to help with fixing that issue. Right now, I'm not sure how decoupling pbs_benchpress from PBSTestSuite and coupling it with CommonTestSuite instead makes any difference.

File: test/fw/ptl/utils/pbs_testsuite.py [hunk 1]
```diff
@@ -52,6 +52,7 @@
 from ptl.utils.pbs_cliutils import CliUtils
 from ptl.utils.pbs_procutils import ProcMonitor
 from ptl.lib.pbs_testlib import *
+from ptl.utils.common_testsuite import *
```
[the remark this answers is not in the document]
Comment (iestockdale) [reply test/fw/ptl/utils/pbs_testsuite.py]: Bhroam, that is a good point.  There should be a document with use cases and the design, with a corresponding discussion in the develoment forum.

**Comment by iestockdale** [comment]
Bhroam, that is a good point. There should be a document with use cases and the design, with a corresponding discussion in the development forum.

**Comment by agrawalravi90** [comment]
This PR has been inactive for 3 months so I'm closing it, please reopen when you are ready with your changes.

**Pull request closed without merging**